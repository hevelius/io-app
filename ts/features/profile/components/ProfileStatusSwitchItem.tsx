import React from "react";
import { StyleSheet, View } from "react-native";
import { useNavigation } from "@react-navigation/native";
import * as pot from "italia-ts-commons/lib/pot";
import { RemoteSwitch } from "../../../components/core/selection/RemoteSwitch";
import { H3 } from "../../../components/core/typography/H3";
import ItemSeparatorComponent from "../../../components/ItemSeparatorComponent";
import NEWPROFILE_ROUTES from "../navigation/routes";

type Props = {
  title: string;
  value: pot.Pot<boolean, Error>;
};

const styles = StyleSheet.create({
  itemRow: {
    flexDirection: "row",
    alignItems: "center",
    padding: 10,
    justifyContent: "space-between"
  }
});

/**
 * This component is used to render a single switch in the profile screen.
 * @param props
 * @returns
 */
const ProfileStatusSwitchItem = (props: Props) => {
  const navigation = useNavigation();
  return (
    <View>
      <View style={styles.itemRow}>
        <H3 testID={"ProfileSwitchItemTitleTestID"}>{props.title}</H3>
        <RemoteSwitch
          testID={"ProfileRemoteSwitchTestID"}
          value={props.value}
          onValueChange={() =>
            navigation.navigate(NEWPROFILE_ROUTES.DELETION.MAIN, {
              screen: NEWPROFILE_ROUTES.DELETION.INFO
            })
          }
        />
      </View>
      <ItemSeparatorComponent noPadded={true} />
    </View>
  );
};

export default ProfileStatusSwitchItem;
