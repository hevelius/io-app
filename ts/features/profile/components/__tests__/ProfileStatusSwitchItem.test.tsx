import React from "react";
import * as pot from "italia-ts-commons/lib/pot";
import { render } from "@testing-library/react-native";
import ProfileStatusSwitchItem from "../ProfileStatusSwitchItem";

type Props = {
  title: string;
  value: pot.Pot<boolean, Error>;
};

const mockedNavigate = jest.fn();

jest.mock("@react-navigation/native", () => {
  const actualNav = jest.requireActual("@react-navigation/native");
  return {
    ...actualNav,
    useNavigation: () => ({
      navigate: mockedNavigate
    })
  };
});

describe("Test ProfileStatusSwitchItem", () => {
  it("should match the snapshot with title and pot.some", () => {
    const props = {
      title: "title",
      value: pot.some(true)
    };
    const component = renderComponent({ ...props });
    expect(component.getByTestId("ProfileSwitchItemTitleTestID")).toBeTruthy();
    expect(component.getAllByText("title")).toBeTruthy();
    expect(component.getByTestId("ProfileRemoteSwitchTestID")).toBeTruthy();
    expect(component).toMatchSnapshot();
  });
  it("should match the snapshot with title and pot.someLoading", () => {
    const props = {
      title: "title",
      value: pot.someLoading(true)
    };
    const component = renderComponent({ ...props });
    expect(component.getByTestId("ProfileSwitchItemTitleTestID")).toBeTruthy();
    expect(component.getAllByText("title")).toBeTruthy();
    expect(component.getByTestId("ProfileRemoteSwitchTestID")).toBeTruthy();
    expect(component).toMatchSnapshot();
  });
});

const renderComponent = ({ title, value }: Props) =>
  render(<ProfileStatusSwitchItem title={title} value={value} />);
