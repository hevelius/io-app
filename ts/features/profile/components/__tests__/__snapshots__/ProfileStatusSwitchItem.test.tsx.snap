// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test ProfileStatusSwitchItem should match the snapshot with title and pot.some 1`] = `
<View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "padding": 10,
      }
    }
  >
    <Text
      color="bluegreyDark"
      font="TitilliumWeb"
      fontStyle={
        Object {
          "fontSize": 18,
        }
      }
      style={
        Array [
          Object {
            "fontSize": 18,
          },
          Object {
            "color": "#17324D",
            "fontFamily": "Titillium Web",
            "fontStyle": "normal",
            "fontWeight": "600",
          },
        ]
      }
      testID="ProfileSwitchItemTitleTestID"
      weight="SemiBold"
      weightColorFactory={[Function]}
    >
      title
    </Text>
    <RCTSwitch
      accessibilityLabel="Select to activate"
      accessibilityRole="switch"
      accessible={true}
      onChange={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      onTintColor="#0073E6"
      style={
        Array [
          Object {
            "height": 31,
            "width": 51,
          },
          Object {
            "marginBottom": -5,
            "marginTop": -5,
          },
        ]
      }
      testID="ProfileRemoteSwitchTestID"
      thumbTintColor="default"
      tintColor="default"
      value={true}
    />
  </View>
  <View
    style={
      Array [
        Object {},
        Object {
          "backgroundColor": "#C9C9C9",
        },
        false,
        Object {
          "height": 0.5,
        },
      ]
    }
  />
</View>
`;

exports[`Test ProfileStatusSwitchItem should match the snapshot with title and pot.someLoading 1`] = `
<View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "padding": 10,
      }
    }
  >
    <Text
      color="bluegreyDark"
      font="TitilliumWeb"
      fontStyle={
        Object {
          "fontSize": 18,
        }
      }
      style={
        Array [
          Object {
            "fontSize": 18,
          },
          Object {
            "color": "#17324D",
            "fontFamily": "Titillium Web",
            "fontStyle": "normal",
            "fontWeight": "600",
          },
        ]
      }
      testID="ProfileSwitchItemTitleTestID"
      weight="SemiBold"
      weightColorFactory={[Function]}
    >
      title
    </Text>
    <View
      style={
        Array [
          Object {},
          Object {
            "width": 51,
          },
        ]
      }
    >
      <ActivityIndicator
        accessibilityLabel="Loading..."
        color="black"
        testID="ProfileRemoteSwitchTestID"
      />
    </View>
  </View>
  <View
    style={
      Array [
        Object {},
        Object {
          "backgroundColor": "#C9C9C9",
        },
        false,
        Object {
          "height": 0.5,
        },
      ]
    }
  />
</View>
`;
